honeycombio/libhoney-py
# Add marker support

## Which problem is this PR solving?

- This PR is part of a SRE interview with honeycomb. I was tasked with enabling functionality to allow customers to create custom markers on graphs. These markers provide annotations in Honeycomb's UI, helping users track significant events like deployments.

## Short description of the changes

- Added a new `markers.py` module that includes a `Marker` class responsible for sending HTTP requests to create markers in a specified dataset.
- Implemented the `create_marker` method, which allows users to specify the dataset, message, and type of marker.
- Added a test script `test_marker.py` for verifying that markers can be created and sent to Honeycomb.

## How to test

- Run the `test_marker.py` script and check the Honeycomb UI to verify that the marker is created successfully.
